tests/lib/config/flat-config-array.js

```diff
@@ -1987,4 +1987,68 @@ describe("FlatConfigArray", () => {
         });
 
     });
+
+    // https://github.com/eslint/eslint/issues/12592
+    describe("Shared references between rule configs", () => {
+
+        it("shared rule config should not cause a rule validation error", () => {
+
+            /* eslint-disable-next-line prefer-const -- need to allow mutation inside FlatConfigArray */
+            let ruleConfig = ["error", {}];
```

Comment:
```suggestion
            const ruleConfig = ["error", {}];
```

I think `let` isn't necessary here.